dataikuapi/utils.py

```diff
@@ -94,3 +94,74 @@ def str_to_bool(s):
                                                 doublequote=True):
                 yield [none_if_throws(caster)(val)
                         for (caster, val) in dku_zip_longest(casters, uncasted_tuple)]
+
+class DSSExtensibleDict(dict):
+    """
+    Utility to define dict-like objects that can be sub-classed.
+    
+    Behaves like dict for most common operations. In particular, 
+    it is possible to update an instance of `:class:`dataikuapi.dss.ml.DSSExtensibleDict` 
+    with either a dict or another instance of `:class:`dataikuapi.dss.ml.DSSExtensibleDict`.
+    
+    Provides an `internal_dict` dict field that is the actual holder of the data.
+    """
+    
+    def __init__(self, orig_dict=None):
+        if orig_dict is None:
+            self.internal_dict = dict()
+        else:
+            self.internal_dict = orig_dict
+
+    def __getitem__(self, key):
+        return self.internal_dict[key]
+
+    def __iter__(self):
+        return self.internal_dict.__iter__()
+
+    def __setitem__(self, key, value):
+        self.internal_dict[key] = value
+
+    def __str__(self):
+        return self.internal_dict.__str__()
```

Comment:
- To make it clear (when exploring in a notebook) that it's not actually a dict and that it also has some additional methods, we may want to do something like `FinalClassName({internal_dict_str})`
- Not sure if we also want to override `__repr__`?

Reply:
So, after some [reading](https://www.journaldev.com/22460/python-str-repr-functions), no need to implement `__str__` if `__repr__` returns a string, so only implement `__repr__`. Thx for pointing this out